spark/common/src/main/scala/org/apache/sedona/spark/SedonaContext.scala

```diff
@@ -65,9 +65,6 @@ object SedonaContext {
     RasterRegistrator.registerAll(sparkSession)
     UdtRegistrator.registerAll()
     UdfRegistrator.registerAll(sparkSession)
-    if (sparkSession.conf.get("spark.sedona.enableParserExtension", "true").toBoolean) {
-      ParserRegistrator.register(sparkSession)
-    }
```

Comment:
NOTE: This introduces a behavior change. Now setting up `spark.sql.extensions` is required for enabling the parser extension. I think this is a reasonable change and I'm OK with it.